src/secret_storage.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper module used to manage interactions with Synapse secrets."""
+
+import logging
+import typing
+from secrets import token_hex
+
+import ops
+from ops.jujuversion import JujuVersion
+
+import actions
+import synapse
+
+PEER_RELATION_NAME = "synapse-peers"
+# Disabling it since these are not hardcoded password
+SECRET_ID = "secret-id"  # nosec
+SECRET_KEY = "secret-key"  # nosec
+
+logger = logging.getLogger(__name__)
+
+
+def _update_peer_data(charm: ops.CharmBase, container: ops.model.Container) -> None:
+    """Update peer data if needed.
+
+    Args:
+        charm: The charm object.
+        container: Synapse container.
+    """
+    # If there is no secret, we use peer relation data
+    # If there is secret, then we update the secret and add the secret id to peer data
+    peer_relation = charm.model.get_relation(PEER_RELATION_NAME)
```

Comment:
You should probably pass the peer relation instead of getting it again as `get_admin_access_token` already gets the relation.
But it's fine testing again the relation as this method could be used elsewhere and forget to check it

Reply:
I'll leave it then just in case.